cardea-data/src/main/java/ca/on/oicr/gsi/cardea/data/ShesmuCase.java

```diff
@@ -117,5 +125,16 @@ public Builder requisitionName(String requisitionName) {
       this.requisitionName = requisitionName;
       return this;
     }
+
+    private Optional<Instant> convertCompletedDate(LocalDate completedDate) {
+      if (completedDate == null) {
+        return Optional.empty();
+      }
+      return Optional.of(
+          ZonedDateTime.of(
+              completedDate,
+              LocalTime.MIDNIGHT,
+              ZoneId.systemDefault()).toInstant());
```

Comment:
This will be ok if Cardea and Shesmu machines are using the same time zone, but the dates will be seen as the day before intended if Shesmu's time zone is ahead of Cardea's. If the system time zone on either is changed (intentionally or not), this can also cause the issue to appear later. How does Shesmu interpret a date (without time) represented as an instant? Ideally, it would treat it as UTC, and then Cardea could explicitly use UTC too

Reply:
Unfortunately, I pulled this from Shesmu's function to create an instant from a date: https://github.com/oicr-gsi/shesmu/blob/a981fd05df363dfbfa186ab2bf385d661a3a7f61/shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/runtime/RuntimeSupport.java#L374

And there are several places in the Shesmu code base which use the system default time zone, so changing this would be more involved: https://github.com/search?q=repo%3Aoicr-gsi%2Fshesmu%20zoneid.systemdefault&type=code

I could modify both Cardea and Shesmu's systemd service files to set timedatectl to a given time zone before startup, which would ensure that they don't get out of sync. It's not the most elegant solution, but would be effective. What do you think?

Reply:
Best solution would be to use a date format without time, but it sounds like Shesmu doesn't accept that and it would take significant effort to change that.

Next best would be to explicitly use UTC on both sides, but changing Shesmu would probably require significant effort and risk breaking things.

It looks like Shesmu is currently using UTC (based on timestamps in UI of shesmu.gsi). If you want to do something to better enforce that, it might be worthwhile. System time zone could still be changed when Shesmu is already running though.

We have more (or at least easier) control over Cardea, so maybe go with the better solution on the Cardea side and explicitly use UTC there instead of system time zone?

Reply:
Alternatively, could use a YYYY,MM,DD Tuple and never touch a time zone

Reply:
Sure, I've enforced UTC in Cardea and I'll propose changing Shesmu's service file in Slack.